network/src/peers/peer/network.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use tokio::{
+    io::{AsyncRead, AsyncReadExt},
+    net::tcp::{OwnedReadHalf, OwnedWriteHalf},
+};
+
+use crate::{NetworkError, Payload};
+
+use super::cipher::Cipher;
+
+// used in integration tests
+#[doc(hidden)]
+pub struct PeerNetwork {
```

Comment:
this name isn't great, as it's just a single connection; maybe a `PeerStream` or a `PeerConnection` would be better here?

Reply:
the intent is more akin to `PeerNetworkHandle` or a `PeerIOHandle` imo. might not be a stream or a connection -- it's intent is to abstract the read/write/IO stuff. will rename